File: iosApp/iosApp/feature/update/UpdateViewState.swift
//
//  UpdateViewState.swift
//  iosApp
//
//  Created by Марк Шавловский on 09.01.2025.
//  Copyright © 2025 orgName. All rights reserved.
//

import Foundation
import shared

struct UpdateViewState {
    let state: UpdateState
}

enum UpdateState {
    case loading
    case success(Link?)
    case error
}

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/ui/screen/update/UpdateViewState.kt
package com.bunbeauty.shared.ui.screen.update

import androidx.compose.runtime.Immutable
import com.bunbeauty.core.model.link.Link
import com.bunbeauty.shared.presentation.base.BaseViewState

@Immutable
data class UpdateViewState(
    val state: State,
) : BaseViewState {
    @Immutable
    sealed interface State {
        data object Loading : State

        data object Error : State

        data class Success(
            val link: Link?,
        ) : State
    }
}
